test/Interop/Cxx/objc-correctness/const-ref-parameter.swift

```diff
@@ -0,0 +1,77 @@
+// REQUIRES: objc_interop
+
+import ConstRefParameter
+
+func testFunction() {
+    let a = OptionsStruct(intOption: 1, floatOption: 2.0)
+
+    let objc = OptionsConsumerObjC(options: a)
+    _ = objc.doOtherThing(withOptions: a)
+    _ = OptionsConsumerObjC.consumer(withOptions: a)
+    _ = OptionsConsumerObjC.doThing(withOptions: a)
+
+    var cxx = OptionsConsumerCxx(a)
+    _ = cxx.doOtherThing(a)
+    _ = OptionsConsumerCxx.build(a)
+    _ = OptionsConsumerCxx.doThing(a)
+}
+
+// RUN: %target-swift-ide-test -print-module -module-to-print=ConstRefParameter -I %S/Inputs -source-filename=x -enable-experimental-cxx-interop -enable-objc-interop | %FileCheck -check-prefix=CHECK-IDE-TEST %s
+
+// CHECK-IDE-TEST: class OptionsConsumerObjC
+// CHECK-IDE-TEST: init(options: OptionsStruct)
+// COM: FIXME: should it be consumer(options:)?
+// CHECK-IDE-TEST: class func consumer(withOptions options: OptionsStruct) -> Self
+// COM: FIXME: should it be doThing(options:)?
+// CHECK-IDE-TEST: class func doThing(withOptions options: OptionsStruct) -> Int32
+// COM: FIXME: should it be doOtherThing(options:)?
+// CHECK-IDE-TEST: func doOtherThing(withOptions options: OptionsStruct) -> Float
+
+// CHECK-IDE-TEST: struct OptionsConsumerCxx
+// CHECK-IDE-TEST: init(_ options: OptionsStruct)
+// CHECK-IDE-TEST: static func build(_ options: OptionsStruct) -> OptionsConsumerCxx
+// CHECK-IDE-TEST: static func doThing(_ options: OptionsStruct) -> Int32
+// CHECK-IDE-TEST: mutating func doOtherThing(_ options: OptionsStruct) -> Float
+
+
+// RUN: %target-swift-frontend -c -enable-experimental-cxx-interop -enable-objc-interop -I %S/Inputs %s -emit-sil -o - | %FileCheck %s
+
+// COM: FIXME: should it be @in_guaranteed OptionsStruct?
```

Comment:
Are these FIXMEs still relevant?

Reply:
Yes. We are still discussing which one of `@in` or `@in_guaranteed` avoid crashes, and also there's this weird behaviour that the initializer (this one) end up being `OptionsStruct` instead of `@in OptionStruct`.

Puyan is working on this and I think we want to try to land something about this in a later PR.

Reply:
What was the resolution of this discussion? I think we want to use `@in_guaranteed`, right? Assuming that's the case, can you create a GH issue and put that in the comment. Then I think we can have one or zero of these FIXMEs. And in that case we know what the bug is so the comment doesn't need to be framed as a question.

Reply:
I think the last decision is try to use `@in_guaranteed`. I can add links to #60601 here to make it clearer.